sklearn/impute.py

```diff
@@ -1144,26 +1146,37 @@ def _get_missing_features_info(self, X):
             imputer_mask = sparse_constructor(
                 (mask, X.indices.copy(), X.indptr.copy()),
                 shape=X.shape, dtype=bool)
+            imputer_mask.eliminate_zeros()
 
-            missing_values_mask = imputer_mask.copy()
-            missing_values_mask.eliminate_zeros()
-            features_with_missing = (
-                np.flatnonzero(np.diff(missing_values_mask.indptr))
-                if missing_values_mask.format == 'csc'
-                else np.unique(missing_values_mask.indices))
+            if self.features in ('missing-only', 'not-constant'):
+                if imputer_mask.format == 'csc':
```

Comment:
this can be achieved with `imputer_mask.getnnz(axis=0)`

Reply:
Why would you make it simple when you can make it complicated :D ?